src/main/java/org/mule/extension/db/internal/domain/connection/datasource/DbDataSourceReferenceConnectionProvider.java

```diff
@@ -29,6 +36,15 @@
     nameRegexpMatcher = DRIVER_FILE_NAME_PATTERN, type = JAR, optional = true)
 public class DbDataSourceReferenceConnectionProvider extends DataSourceReferenceConnectionProvider {
 
+  private final Map<String, Map<Integer, ResolvedDbType>> resolvedDbTypesCache = new ConcurrentHashMap<>();
 
+  @Override
+  protected DbConnection createDbConnection(Connection connection) throws Exception {
+    if (connection.getMetaData().getDatabaseProductName().equals("Oracle")) {
```

Comment:
Should we introduce a new Helper class to distinguish vendors given a connection?

Reply:
Yes, that's a good idea. What if I add a `handles` static method to `OracleDbConnection`?

The code would then be...

```
    if (OracleDbConnection.handles(connection)) {
      return new OracleDbConnection(connection, super.resolveCustomTypes(), resolvedDbTypesCache);
    } else {
      return super.createDbConnection(connection);
    }
```